File: InterSectionOfTwoArraysII/InsecOfTwoArrII.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        arr = []
        for i in nums1:
            if i in nums2:
                arr.append(i)
                nums2.remove(i)
        return arr


class Solution(object):
    def intersect(self, nums1, nums2):
        nums1.sort()
        nums2.sort()
        arr = []
        i, j = 0, 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] < nums2[j]:
                i += 1
            elif nums2[j] < nums1[i]:
                j += 1
            else:
                arr.append(nums1[i])
                i += 1
                j += 1
        return arr

File: InterSectionOfTwoArraysII/InsecOfTwoArrII.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function (nums1, nums2) {
  let arr = [];
  nums1.map((num1, i) => {
    if (nums2.includes(num1)) {
      arr.push(num1);
      nums2.splice(nums2.indexOf(num1), 1);
    }
  });
  return arr;
};


/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
 var intersect = function (nums1, nums2) {
    nums1.sort((a, b) => a - b)
    nums2.sort((a, b) => a - b)
    let arr = []
    let i = 0, j = 0
    while(i < nums1.length && j < nums2.length) {
        if (nums1[i] < nums2[j]) i++
        else if (nums2[j] < nums1[i]) j++
        else {
            arr.push(nums1[i])
            i++
            j++
        }
    }
    return arr
};
